gdsfactory/path.py

```diff
@@ -136,13 +112,50 @@ def __add__(self, path: npt.NDArray[np.floating[Any]] | Path) -> Path:
         return new.append(path)
 
     @property
-    def bbox(self) -> npt.NDArray[np.floating[Any]]:
-        """Returns the bounding box of the Path."""
-        bbox = [
-            (np.min(self.points[:, 0]), np.min(self.points[:, 1])),
-            (np.max(self.points[:, 0]), np.max(self.points[:, 1])),
-        ]
-        return np.array(bbox)
+    def kcl(self) -> kf.KCLayout:
+        return gf.kcl
+
+    def transform(
+        self,
+        trans: kdb.Trans | kdb.DTrans | kdb.ICplxTrans | kdb.DCplxTrans,
+        /,
+    ) -> Any:
+        if isinstance(trans, kdb.DCplxTrans):
+            trans_ = trans
+        elif isinstance(trans, kdb.DTrans):
+            trans_ = kdb.DCplxTrans(trans)
+        elif isinstance(trans, kdb.Trans):
+            trans_ = kdb.DCplxTrans(trans.to_dtype(gf.kcl.dbu))
+        else:
+            trans_ = trans.to_itrans(gf.kcl.dbu)
+        angle_rad = np.radians(trans_.angle)
+        cos_angle = np.cos(angle_rad)
+        sin_angle = np.sin(angle_rad)
```

Comment:
Can you check the speed difference if you check whether there is a big speed difference if we check for no rotation vs  there being a rotation?